src/client/Microsoft.Identity.Client/Utils/DateTimeHelpers.cs

```diff
@@ -63,15 +63,39 @@ public static long GetDurationFromWindowsTimestamp(string windowsTimestampInFutu
             return (long)unixTimestamp - CurrDateTimeInUnixTimestamp();
         }
 
-        public static long GetDurationFromNowInSeconds(string unixTimestampInFuture)
+        public static long GetDurationFromNowInSeconds(string dateTimeStamp)
         {
-            if (string.IsNullOrEmpty(unixTimestampInFuture))
+            if (string.IsNullOrEmpty(dateTimeStamp))
             {
                 return 0;
             }
 
-            long expiresOnUnixTimestamp = long.Parse(unixTimestampInFuture, CultureInfo.InvariantCulture);
-            return expiresOnUnixTimestamp - CurrDateTimeInUnixTimestamp();
+            // First, try to parse as Unix timestamp (number of seconds since epoch)
+            if (long.TryParse(dateTimeStamp, out long expiresOnUnixTimestamp))
+            {
+                return expiresOnUnixTimestamp - DateTimeHelpers.CurrDateTimeInUnixTimestamp();
+            }
+
+            // Try parsing as ISO 8601 
+            if (DateTimeOffset.TryParse(dateTimeStamp, null, DateTimeStyles.RoundtripKind, out DateTimeOffset expiresOnDateTime))
+            {
+                return (long)(expiresOnDateTime - DateTimeOffset.UtcNow).TotalSeconds;
+            }
+
+            // Try RFC 1123 format 
+            if (DateTimeOffset.TryParseExact(dateTimeStamp, "R", CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal, out expiresOnDateTime))
+            {
+                return (long)(expiresOnDateTime - DateTimeOffset.UtcNow).TotalSeconds;
+            }
+
+            // Try parsing Unix timestamp in milliseconds 
+            if (long.TryParse(dateTimeStamp, out long expiresOnMillisTimestamp) && dateTimeStamp.Length > 10)
```

Comment:
Does this code path actually get hit in the known product code? I don't think we should support any kind of format.